packages/dataprovider/src/types.ts

```diff
@@ -10,6 +8,8 @@ import {
   UPDATE,
 } from "react-admin";
 
+import { DocumentNode } from "graphql";
+
```

Comment:
This should probably be added below the declaration for `QueryDialect`, but github doesn't let me put a comment there.

```suggestion
import { Resource } from "./constants/interfaces";

export type MutationFetchType = typeof CREATE | typeof DELETE | typeof UPDATE;

export type MutationOperationNameMap = {
  [K in MutationFetchType]: (resource: Resource) => string;
};

export type MutationOperationNames = Partial<
  Record<QueryDialect, MutationOperationNameMap>
>;
```

It would also make sense to break `FetchType` up a bit so that it can leverage the `MutationFetchType`:
```ts
export type QueryFetchType =
  | typeof GET_LIST
  | typeof GET_MANY
  | typeof GET_MANY_REFERENCE
  | typeof GET_ONE;

export type FetchType = QueryFetchType | MutationFetchType;
```

Reply:
awesome catch!